src/core/execute.c

```diff
@@ -3222,6 +3237,7 @@ void exec_context_done(ExecContext *c) {
         c->environment = strv_free(c->environment);
         c->environment_files = strv_free(c->environment_files);
         c->pass_environment = strv_free(c->pass_environment);
+        c->unset_environment = strv_free(c->pass_environment);
```

Comment:
This line seems to lead to a memory leak
```sh
==17636==
==17636== 22 (16 direct, 6 indirect) bytes in 1 blocks are definitely lost in loss record 2 of 8
==17636==    at 0x482E17C: malloc (in /usr/lib/valgrind/vgpreload_memcheck-x86-linux.so)
==17636==    by 0x48304A0: realloc (in /usr/lib/valgrind/vgpreload_memcheck-x86-linux.so)
==17636==    by 0x23CFA4: strv_extend_strv (strv.c:216)
==17636==    by 0x14D09E: config_parse_unset_environ (load-fragment.c:2300)
==17636==    by 0x1DECC2: next_assignment (conf-parser.c:151)
==17636==    by 0x1DF393: parse_line (conf-parser.c:270)
==17636==    by 0x1DF8E8: config_parse (conf-parser.c:373)
==17636==    by 0x154844: load_from_path (load-fragment.c:4342)
==17636==    by 0x154A52: unit_load_fragment (load-fragment.c:4381)
==17636==    by 0x158057: service_load (service.c:690)
==17636==    by 0x121818: unit_load (unit.c:1322)
==17636==    by 0x137AAE: manager_dispatch_load_queue (manager.c:1566)
==17636==
```

Reply:
ouch! indeed!